x-pack/plugins/cases/server/client/cases/search.ts

```diff
@@ -57,39 +63,61 @@ export const find = async (
       licensingService.notifyUsage(LICENSING_CASE_ASSIGNMENT_FEATURE);
     }
 
+    /**
+     * Verify if custom field type is filterable
```

Comment:
nit: I think this comment is a bit misleading.

That is all done inside `validateSearchCasesCustomFields` and a couple of other things are verified.